packages/@expo/cli/src/run/android/resolveGradlePropsAsync.ts

```diff
@@ -60,5 +62,7 @@ async function getConnectedDeviceABIS(
   }
 
   const abis = await getDeviceABIsAsync(device);
-  return abis.filter((abi, i, arr) => arr.indexOf(abi) === i).join(',');
+
+  const validAbis = abis.filter((abi) => VALID_ARCHITECTURES.includes(abi));
+  return validAbis.filter((abi, i, arr) => arr.indexOf(abi) === i).join(',');
```

Comment:
What happens when there are no valid abis detected, do we still pass the gradle prop to RN?

Reply:
No, we have a check if it's falsey  later so we won't add it.